data-collection/data-collection/View Controllers/SegmentedViewController.swift

```diff
@@ -0,0 +1,254 @@
+//// Copyright 2019 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+
+// This class is intended to be subclassed and is done so in `RichPopupViewController`.
+class SegmentedViewController: UIViewController {
+    
+    // MARK: Views
+    
+    private let effectsSegmentedControlContainerView: UIVisualEffectView = {
+        
+        // Build visual effect view with blur view.
+        let visualEffect = UIBlurEffect(style: .extraLight)
+        let visualEffectView = UIVisualEffectView(effect: visualEffect)
+        visualEffectView.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Build 1pt spacer view.
+        let spacer = UIView(frame: .zero)
+        spacer.backgroundColor = UIColor(white: 0.75, alpha: 1.0)
+        spacer.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Add and constrain spacer.
+        visualEffectView.contentView.addSubview(spacer)
+        
+        var constraints = [NSLayoutConstraint]()
+        constraints.append( spacer.leadingAnchor.constraint(equalTo: visualEffectView.contentView.leadingAnchor) )
+        constraints.append( spacer.trailingAnchor.constraint(equalTo: visualEffectView.contentView.trailingAnchor) )
+        constraints.append( spacer.bottomAnchor.constraint(equalTo: visualEffectView.contentView.bottomAnchor) )
+        constraints.append( spacer.heightAnchor.constraint(equalToConstant: (1.0/UIScreen.main.scale)) )
+        NSLayoutConstraint.activate(constraints)
+        
+        return visualEffectView
+    }()
+    
+    let segmentedControl: UISegmentedControl = {
+        let segmentedControl = UISegmentedControl(frame: .zero)
+        segmentedControl.translatesAutoresizingMaskIntoConstraints = false
+        segmentedControl.addTarget(self, action: #selector(segmentedControlValueDidChange(_:)), for: .valueChanged)
+        return segmentedControl
+    }()
+    
+    // MARK: Public Interface
+    
+    /// Refreshes titles for each segment with title of the cooresponding controller.
+    func refreshSegmentTitles() {
+        
+        for (index, child) in childrenViewControllers.enumerated() {
+            segmentedControl.setTitle(child.title, forSegmentAt: index)
+        }
+    }
+    
+    /// Should override and provide array of segue identifiers.
+    public func segmentedViewControllerChildIdentifiers() -> [String] { return [] }
+    
+    private var childrenIdentifiers: [String]!
```

Comment:
Consider whether this really needs to be optional.

```suggestion
    private var childrenIdentifiers = [String]()
```